ggml-cuda.cu

```diff
@@ -6023,18 +6049,18 @@ static cudaError_t ggml_cuda_cpy_tensor_2d(
     const char * x = src_ptr + i1_low*nb1 + i2*nb2 + i3*nb3;
     if (nb0 == ts && nb1 == ts*ne0/bs) {
         return cudaMemcpyAsync(dst_ptr, x, i1_diff*nb1, kind, stream);
-    } else if (nb0 == ts) {
+    }
+    if (nb0 == ts) {
         return cudaMemcpy2DAsync(dst_ptr, ts*ne0/bs, x, nb1, ts*ne0/bs, i1_diff, kind, stream);
-    } else {
-        for (int64_t i1 = 0; i1 < i1_diff; i1++) {
-            const void * rx = (const void *) ((const char *) x + i1*nb1);
-            void * rd = (void *) (dst_ptr + i1*ts*ne0/bs);
-            // pretend the row is a matrix with cols=1
-            cudaError_t r = cudaMemcpy2DAsync(rd, ts/bs, rx, nb0, ts/bs, ne0, kind, stream);
-            if (r != cudaSuccess) return r;
-        }
-        return cudaSuccess;
     }
+    for (int64_t i1 = 0; i1 < i1_diff; i1++) {
+        const void * rx = (const void *) ((const char *) x + i1*nb1);
+        void * rd = (void *) (dst_ptr + i1*ts*ne0/bs);
+        // pretend the row is a matrix with cols=1
+        cudaError_t r = cudaMemcpy2DAsync(rd, ts/bs, rx, nb0, ts/bs, ne0, kind, stream);
+        if (r != cudaSuccess) { return r; }
+    }
+    return cudaSuccess;
```

Comment:
Hm, what happened with the `else`s here - is this intended to remove them?

Not 100%, but looks like the original version was correct?

@KerfuffleV2

Reply:
It was intended. The original version basically was like:

```c++
  if (a) {
    x();
    return 1;
  } else if (b) {
    y();
    return 2;
  } else {
    return 3;
  }
  // function ends
```

Since each branch of the conditional lead to a return, the "else"s were redundant and it's functionally identical to:


```c++
  if (a) {
    x();
    return 1;
  } 
  if (b) {
    y();
    return 2;
  }
  return 3;
  // function ends
}
```

This wasn't something I did unilaterally, `clang-tidy` complained about it based on this repo's `clang-tidy` configuration.